contracts/DecentHats.sol

```diff
@@ -0,0 +1,436 @@
+     * hat's smart account must be the one call to `withdraw-` on the Sablier contract, setting the recipient arg to its wearer.
+     *
+     * @dev Role hat creation, minting, smart account creation and stream creation are handled here in order
+     * to avoid a race condition where not more than one active proposal to create a new role can exist at a time.
+     * See: https://github.com/decentdao/decent-interface/issues/2402
+     */
+    function createRoleHat(CreateRoleHatParams calldata params) external {
+        _createHatAndAccountAndMintAndStreams(
+            params.hatsProtocol,
+            params.registry,
+            params.topHatAccount,
+            params.hatsAccountImplementation,
+            params.adminHatId,
+            params.hat
+        );
+
+        params.hatsProtocol.transferHat(
+            params.topHatId,
+            address(this),
+            msg.sender
+        );
+    }
+
+    /**
+     * @notice Creates a new termed role hat and any streams on it.
+     *
+     * @notice This contract should be enabled a module on the Safe for which the role is to be created, and disable after.
+     *
+     * @dev In order for the module to be able to create hats on behalf of the Safe, the Safe must first
+     * transfer its top hat to this contract. This function transfers the top hat back to the Safe after
+     * creating the role hat.
+     *
+     * @dev The function simply calls `_createTermedHatAndAccountAndMintAndStreams` and then transfers the top hat back to the Safe.
+     *
+     * @dev Stream funds on Termed Roles are targeted directly at the nominated wearer.
+     * The wearer should directly call `withdraw-` on the Sablier contract.
+     *
+     * @dev Termed Role hat creation, minting, and stream creation are handled here in order
+     * to avoid a race condition where not more than one active proposal to create a new termed role can exist at a time.
+     * See: https://github.com/decentdao/decent-interface/issues/2402
+     */
+    function createTermedRoleHat(
+        CreateTermedRoleHatParams calldata params
+    ) external {
+        _createTermedHatAndAccountAndMintAndStreams(
+            params.hatsProtocol,
+            params.topHatAccount,
+            params.hatsModuleFactory.createHatsModule( // Create election module and set as eligibility
+                params.hatsElectionEligibilityImplementation,
+                params.hatsProtocol.getNextId(params.adminHatId),
+                abi.encode(params.topHatId, uint256(0)), // [BALLOT_BOX_ID, ADMIN_HAT_ID]
+                abi.encode(params.hat.termedParam.termEndDateTs),
+                uint256(SALT)
+            ),
+            params.adminHatId,
+            params.hat
+        );
+
+        params.hatsProtocol.transferHat(
+            params.topHatId,
+            address(this),
+            msg.sender
+        );
+    }
```

Comment:
Oh look this function exists too!

@Da-Colon @DarksightKellar see my previous comment in this PR. Same applies here.

Reply:
Let's talk about this today.

Reply:
ready when you are 🫡